lib/src/block_syntaxes/blockquote_syntax.dart

```diff
@@ -97,3 +97,65 @@ class BlockquoteSyntax extends BlockSyntax {
     return Element('blockquote', children);
   }
 }
+
+/// Parses GitHub Alerts blocks.
+///
+/// See also: https://docs.github.com/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts
+class AlertBlockSyntax extends BlockSyntax {
+  @override
+  RegExp get pattern => alertPattern;
+
+  const AlertBlockSyntax();
+
+  @override
+  List<Line> parseChildLines(BlockParser parser) {
+    return const BlockquoteSyntax().parseChildLines(parser);
+  }
+
+  @override
+  Node parse(BlockParser parser) {
+    // Parse the alert type from the first line.
+    final type =
+        pattern.firstMatch(parser.current.content)!.group(1)!.toLowerCase();
+
+    final childLines = parseChildLines(parser);
+    // Until we've parse all the child lines, we can't actually know if this is
+    // a blockquote containing `[!note]` or if this is an alert-block.
+    //
+    // This is because `> [!note]` is not a valid alert-block!
```

Comment:
Can you explain **why** this is not a valid alert block? because it is empty?

Reply:
I don't know, that's how it behaves on github :D

Reply:
But is it the emptiness that makes it invalid, or some other property?

Reply:
# Example of the empty case

```md
-----------
> [!note]
-----------
```

-----------
> [!note]
-----------


# Example of the non-empty case

```md
-----------
> [!note]
> Hello world
-----------
```

-----------
> [!note]
> Hello world
-----------

# Conclusion

If we have `> [!note]` it's not a special block. But if it's:
 * Followed by text that wants to belong to the previous paragraph, then it is. OR
 * Followed by lines like `> ...` indicating that whatever is there belongs inside the blockquote, then it also is special.

Otherwise, it's not special, just a normal blockquote.

Notice that `> foo\nbar` is the same as `> foo\n> bar`, the line that follows is considered to be a paragraph continuation of the previous line, so the blockquote continues.

So yes, as far as I can see, it's emptiness that decides it. Even if you have `> [!note]\n> ` it'll still not be rendered as a special alert-block.